examples/web-authn-oauth.py

```diff
@@ -46,20 +46,24 @@ class MyOAuthSecuredApplication(asab.Application):
 
 		oauth_methods = [asab.web.authn.oauth.GitHubOAuthMethod()]
 
+		# Create cache for identities
+
+		oauth_identity_cache = asab.web.authn.oauth.OAuthIdentityCache(self, methods=oauth_methods)
+
 		# Add middleware for authentication via oauth2
 		container.WebApp.middlewares.append(
 			asab.web.authn.authn_middleware_factory(
 				self,
 				"oauth2client",
-				methods=oauth_methods,
+				identity_cache=oauth_identity_cache,
```

Comment:
Definitively not, cache is an implementation detail and we are not going to expose this to an user.
If you cannot hide that, we need to introduce an OAuthClientService that will have to be initialized (by methods) and cache will be part of it. 

Middleware can refer to that service by service id and/or direct service object reference.
`asab.web.authn.oauth.OAuthForwarder` will become a method of that service.